litt/cache/cached_table.go

```diff
@@ -114,17 +119,28 @@ func (c *cachedTable) CacheAwareGet(
 	onlyReadFromCache bool,
 ) (value []byte, exists bool, hot bool, err error) {
 
+	if c.metrics != nil {
+		start := time.Now()
+		defer func() {
+			if exists && value != nil {
+				c.metrics.ReportReadOperation(c.Name(), time.Since(start), uint64(len(value)), hot)
+			}
+		}()
+	}
+
 	stringKey := util.UnsafeBytesToString(key)
 
 	value, exists = c.writeCache.Get(stringKey)
 	if exists {
 		// The value was recently written
-		return value, true, true, nil
+		hot = true
+		return value, exists, hot, nil
 	} else {
 		value, exists = c.readCache.Get(stringKey)
 		if exists {
 			// The value was recently read
-			return value, true, true, nil
+			hot = true
+			return value, exists, hot, nil
 		}
 	}
 
```

Comment:
There is still one return case where you manually specify bools, instead of using `exists` and `hot`. Is that intentional?

Reply:
There was a method to the madness, but since it wasn't obvious to the reader, probably not a good pattern. I now always return the variables in this function.